server/pulp/server/managers/consumer/applicability.py

```diff
@@ -118,29 +120,52 @@ def regenerate_applicability_for_repos(repo_criteria):
         repo_criteria.fields = ['id']
         repo_ids = [r.repo_id for r in model.Repository.objects.find_by_criteria(repo_criteria)]
 
+        task_group_id = uuid4()
+
         for repo_id in repo_ids:
             # Find all existing applicabilities for given repo_id. Setting batch size of 5 ensures
             # the MongoDB cursor does not time out. See https://pulp.plan.io/issues/998#note-6 for
             # more details.
             existing_applicabilities = RepoProfileApplicability.get_collection().find(
-                {'repo_id': repo_id}).batch_size(5)
+                {'repo_id': repo_id})
+            counter = 0
+            existing_applicabilities_batch = []
             for existing_applicability in existing_applicabilities:
+                existing_applicabilities_batch.append(existing_applicability)
+                counter += 1
+                if counter == 100:
+                    regenerate_applicability_for_repos_task.apply_async(repo_id,
+                                                                        existing_applicabilities_batch,
+                                                                        group_id=task_group_id)
+                    counter = 0
+                    existing_applicabilities_batch = []
+
+            if len(existing_applicabilities_batch) > 0:
+                # Dispatch the remaining batch
+                task_kwargs = {"group_id": task_group_id}
+                regenerate_applicability_for_repos_task.apply_async((repo_id,
+                                                                    existing_applicabilities_batch),**task_kwargs)
+        return task_group_id
+
+    @staticmethod
+    def regenerate_applicability_for_repos_task(repo_id, existing_applicabilities, group_id=None):
```

Comment:
need a doc block. For tasks, it's preferable to pass IDs instead of objects, so we don't have to serialize objects and stuff them in an AMQP message. A list of unique IDs would be ideal for the ``existing_applicabilities`` parameter.